### PR TITLE
fix: crashing message composer input [WPB-8727]

### PR DESCRIPTION
**Pull request by saleniuk**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-8727" title="WPB-8727" target="_blank">BugWPB-8727</a>  [Android] Crash when using tapping on input field while trying to type a message
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

The app is crashing sometimes when clicking on the message composer input.

### Causes (Optional)

Changing `readOnly` parameter probably messes something up with the inner text field state so that it ends up with `java.lang.IllegalStateException: LayoutCoordinate operations are only valid when isAttached is true`.

### Solutions

Getting rid of `readOnly` is not possible because it was introduced some time ago to fix some other issues that we had with message composer and keyboard.
New `BasicTextField2` resolves multiple issues that the previous generation of text input had, including this one. `BasicTextField2` has been renamed to regular `BasicTextField` in newest compose library; the difference is that new one takes `TextFieldState` instead of `TextFieldValue` or `String`. 
We decided to use this new version of text field for the message composer. In order to do that, a new hybrid `WireTextField2` is created which allows us to use the new generation of text input but also to still pass `TextFieldValue` and `onValueChanged` callback to make this refactor as minimally invasive as possible. This `WireTextField2` uses `StateSyncingModifier` which is based on compose [`BasicTextFieldWithValueOnValueChangeSample`](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:compose/foundation/foundation/samples/src/main/java/androidx/compose/foundation/samples/BasicTextFieldValueSample.kt) which allows to synchronise between `TextFieldState` and `TextFieldValue` with `onValueChanged` callback.
It's needed to update the `compose-foundation` library to `1.7.0-alpha05`, from which `BasicTextField2` has been marked as stable and still older `BasicTextField` is available. In versions prior to chosen one, including current newest release version - `1.6.7`, `BasicTextField2` has some bad issues with the keyboard (https://issuetracker.google.com/issues/339171226) and interaction source (https://issuetracker.google.com/issues/327665606). To use this version, `rememberRipple()` needs to be updated to `ripple()` and `AnchoredDraggableState` also requires `decayAnimationSpec`.

#### How to Test

Open conversation screen and click on message composer input right after navigating, preferably on Graphene OS.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [x] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [x] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [4687](https://10.10.124.134/job/AR-build-pipeline/4687/) **failed**.

**Comment by AndroidBob** [comment]
Build [4688](https://10.10.124.134/job/AR-build-pipeline/4688/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextField2.kt [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+            inputTransformation = InputTransformation.maxLength(maxTextLength),
+            scrollState = scrollState,
+            readOnly = readOnly,
+            enabled = state !is WireTextFieldState.Disabled,
+            cursorBrush = SolidColor(MaterialTheme.colorScheme.primary),
+            interactionSource = interactionSource,
+            modifier = Modifier
+                .fillMaxWidth()
+                .background(color = colors.backgroundColor(state).value, shape = shape)
+                .border(width = 1.dp, color = colors.borderColor(state, interactionSource).value, shape = shape)
```
Comment (Garzas) [inline app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextField2.kt]: use dimensions

File: app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextField2.kt [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+            !descriptionText.isNullOrEmpty() -> descriptionText
+            else -> String.EMPTY
+        }
+        AnimatedVisibility(visible = bottomText.isNotEmpty()) {
+            Text(
+                text = bottomText,
+                style = MaterialTheme.wireTypography.label04,
+                textAlign = TextAlign.Start,
+                color = colors.descriptionColor(state).value,
+                modifier = Modifier.padding(top = 4.dp)
```
Comment (Garzas) [inline app/src/main/kotlin/com/wire/android/ui/common/textfield/WireTextField2.kt]: use dimensions

**Review by Garzas — APPROVED** [review]
Excellent job 🎸  small comments about dimensions 💪

**Comment by AndroidBob** [comment]
Build [4690](https://10.10.124.134/job/AR-build-pipeline/4690/) **failed**.

**Comment by AndroidBob** [comment]
Build [4691](https://10.10.124.134/job/AR-build-pipeline/4691/) **failed**.

**Review by ohassine — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [4698](https://10.10.124.134/job/AR-build-pipeline/4698/) **failed**.

**Comment by yamilmedina** [comment]
It is failing in CI because of

```
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/calling/controlbuttons/CameraButton.kt:25:34 Packages cannot be imported
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/calling/controlbuttons/CameraButton.kt:65:34 Unresolved reference: ripple
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/calling/controlbuttons/CameraFlipButton.kt:24:34 Packages cannot be imported
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/calling/controlbuttons/CameraFlipButton.kt:49:34 Unresolved reference: ripple
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/calling/controlbuttons/DeclineButton.kt:24:34 Packages cannot be imported
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/calling/controlbuttons/DeclineButton.kt:48:34 Unresolved reference: ripple
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/calling/controlbuttons/MicrophoneButton.kt:25:34 Packages cannot be imported
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/calling/controlbuttons/MicrophoneButton.kt:52:34 Unresolved reference: ripple
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/calling/controlbuttons/SpeakerButton.kt:25:34 Packages cannot be imported
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/calling/controlbuttons/SpeakerButton.kt:52:34 Unresolved reference: ripple
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/common/AppExtensions.kt:24:34 Packages cannot be imported
e: file:///home/megazord/builder-agent/android-reloaded-builder/workspace/AR-build-pipeline@2/app/src/main/kotlin/com/wire/android/ui/common/AppExtensions.kt:65:22 Unresolved reference: ripple
```

**Comment by AndroidBob** [comment]
Build [4709](https://10.10.124.134/job/AR-build-pipeline/4709/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9637898)(9637898)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9637898)(9637898)-staging-compat.apk)

**Pull request merged**